packages/mask/src/plugins/Referral/SNSAdaptor/BuyToFarm.tsx

```diff
@@ -0,0 +1,225 @@
+import { useCallback, useState } from 'react'
+import { useAsync } from 'react-use'
+import type { Web3 } from '@masknet/web3-shared-evm'
+import { useAccount, useChainId, useWeb3 } from '@masknet/plugin-infra/web3'
+import { makeStyles, useCustomSnackbar } from '@masknet/theme'
+import { useRemoteControlledDialog } from '@masknet/shared-base-ui'
+import { Typography, Box, Tab, Tabs, Grid, Divider } from '@mui/material'
+import { TabContext, TabPanel } from '@mui/lab'
+import { v4 as uuid } from 'uuid'
+import { EMPTY_LIST } from '@masknet/shared-base'
+import { NetworkPluginID } from '@masknet/web3-shared-base'
+
+import { useI18N } from '../locales'
+import { PluginReferralMessages, SelectTokenUpdated, ReferralRPC } from '../messages'
+import { PluginTraderMessages } from '../../Trader/messages'
+
+import { getRequiredChainId } from '../helpers'
+import { singAndPostProofOfRecommendationWithReferrer } from './utils/proofOfRecommendation'
+import { MASK_REFERRER, SWAP_CHAIN_ID } from '../constants'
+import { TabsReferAndBuy, TransactionStatus, PageInterface, PagesType, FungibleTokenDetailed } from '../types'
+import type { Coin } from '../../Trader/types'
+
+import ActionButton from '../../../extension/options-page/DashboardComponents/ActionButton'
+import { WalletConnectedBoundary } from '../../../web3/UI/WalletConnectedBoundary'
+import { ChainBoundary } from '../../../web3/UI/ChainBoundary'
+import { MyRewards } from './MyRewards'
+import { TokenSelectField } from './shared-ui/TokenSelectField'
+import { RewardDataWidget } from './shared-ui/RewardDataWidget'
+import { SponsoredFarmIcon } from './shared-ui/icons/SponsoredFarm'
+
+import { useTabStyles } from './styles'
+
+const useStyles = makeStyles()((theme) => ({
+    container: {
+        flex: 1,
+        height: '100%',
+    },
+    tab: {
+        maxHeight: '100%',
+        height: '100%',
+        overflow: 'auto',
+        padding: theme.spacing(3, 0),
+    },
+    tabs: {
+        width: '288px',
+    },
+    subtitle: {
+        margin: '12px 0 24px',
+    },
+    typeNote: {
+        marginBottom: '24px',
+        '& b': {
+            margin: '0 4px 0 8px',
+            fontWeight: 600,
+        },
+    },
+}))
+
+export function BuyToFarm(props: PageInterface) {
+    const t = useI18N()
+    const { classes } = useStyles()
+    const { classes: tabClasses } = useTabStyles()
+    const currentChainId = useChainId()
+    const requiredChainId = getRequiredChainId(currentChainId)
+    const web3 = useWeb3()
```

Comment:
```suggestion
    const web3 = useWeb3(NetworkPluginID.PLUGIN_EVM)
```

Reply:
@guanbinrui I added this NetworkPluginID.PLUGIN_EVM to useWeb3, useChanId and useAccount hooks in the next commit.
Could you check it please?